### PR TITLE
[Bugfix][LoRA] Skip no-LoRA tokens (index -1) in CPU torch bgmv ops

### PR DESCRIPTION
**Pull request by huthvincent**
## Purpose

`token_lora_indices` marks base-model (no-LoRA) tokens with the sentinel `-1`
(see `PunicaWrapperBase`; the value is documented as "an index of -1 means no
lora should be applied"). The Triton LoRA kernels honor it by early-exiting
(`if lora_id == -1: return` in `lora_shrink_op`/`lora_expand_op`).

The pure-PyTorch fallback ops used on the **CPU platform**
(`vllm/lora/ops/torch_ops/lora_ops.py`) instead did:

```python
selected_loras = lora_b_weights[lora_indices_tensor]
```

For a `-1` token this selects `lora_b_weights[-1]` — the **last stacked
adapter** — instead of skipping. So in a mixed batch (e.g. the default
`max_loras=1` with one adapter loaded), a base-model request batched next to a
LoRA request silently gets the adapter's delta applied to its logits /
embeddings — wrong output, no error. This affects `bgmv_shrink`, `bgmv_expand`
and `bgmv_expand_slice`, and therefore the `sgmv_*` prefill ops that delegate to
them.

**Fix:** clamp the gather index to `0` and zero the LoRA delta for `-1` tokens
(`masked_fill`), mirroring the Triton early-exit. For valid indices the behavior
is bit-for-bit unchanged (`clamp` is a no-op, the mask is all-False), so the
existing kernel-parity tests are unaffected.

## Test Plan

Added `test_torch_ops_bgmv_skips_no_lora_tokens` to
`tests/lora/test_punica_ops.py`: a 2-token batch `[LoRA-0, no-LoRA(-1)]` run
through `bgmv_shrink` / `bgmv_expand` / `bgmv_expand_slice`, asserting the `-1`
token receives a zero delta and the valid token receives the correct delta.

```bash
pytest tests/lora/test_punica_ops.py::test_torch_ops_bgmv_skips_no_lora_tokens
pytest tests/lora/test_punica_ops.py       # full parity suite, no regressions
```

## Test Result

Verified on 1× NVIDIA H200 (torch 2.11.0).

```
# new test, BEFORE fix (base-model token wrongly modified):
>   assert torch.count_nonzero(y[1]) == 0
E   assert tensor(4) == 0
1 failed

# new test, AFTER fix:
1 passed

# full file (torch-vs-triton parity tests), AFTER fix:
4274 passed
```

`pre-commit run --files ...` (ruff, ruff-format, mypy, typos, SPDX) passes.

---

*AI assistance (Claude) was used to develop this change. The submitter has
reviewed every changed line and run the tests above.*

### CODE REVIEW

**Comment by huthvincent** [comment]
Gentle follow-up. I re-checked today: the -1 gather is still on current main (`lora_ops.py` L31/L74/L119, untouched since Oct 2025), and this patch still applies cleanly. #49656 (@abrudner) fixes the same bug with a larger change that also skips the GEMM for no-LoRA rows and fuses `add_shrink`; abrudner kindly offered to defer to this smaller diff. One option is to land this minimal fix first and let #49656 rebase as a perf follow-up, but I'm equally happy to close this in favor of #49656 if you prefer that direction. @bigPYJ1151 @jeejeelee whenever you have a moment, no rush. Thanks!
